api/libs/scenario-cost-surface/src/cost-surface-file-cache.api.entity.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  Column,
+  CreateDateColumn,
+  Entity,
+  Index,
+  PrimaryColumn,
+  UpdateDateColumn,
+} from 'typeorm';
+
+export enum ArtifactType {
+  CostTemplate = 'CostTemplate',
+}
+
+@Entity('cost_surface_file_cache')
+@Index(['scenarioId', 'artifactType'])
+export class CostSurfaceFileCache {
+  @PrimaryColumn({
+    generated: 'uuid',
+    type: 'uuid',
+  })
+  id!: string;
+
+  @Column()
+  scenarioId!: string;
+
+  @Column({ type: 'int4', nullable: true })
+  artifact?: number;
+
+  @Column({
+    type: 'enum',
+    enum: ArtifactType,
+  })
+  artifactType!: ArtifactType;
+
+  @Column()
+  contentType!: string;
+
+  @CreateDateColumn({ type: 'timestamp without time zone' })
+  createdAt!: Date;
+
+  @UpdateDateColumn({ type: 'timestamp without time zone' })
+  lastModifiedAt!: Date;
+}
+
+export const createCostSurfaceFileCacheFields = [
+  'scenarioId',
+  'artifactType',
+  'contentType',
+] as const;
```

Comment:
is it an entity responsibility to decide which files are under cache? 🤔

Reply:
the const is defining only which fields are needed to create an instance

Reply:
just for a sake of security of typing, shouldn't they relate to actual fields?

Reply:
it should be ok as https://github.com/Vizzuality/marxan-cloud/pull/228/files/14076746b22f7c4e320e8e6473503343e85b8ba3#diff-0298940b2ab35a3d32a4536133f253445f3c2ce55b768011266c254e8fd80596R42 would bark if we used this array including in it elements which don't match props of `CostSurfaceFileCache`

Reply:
I've added a `type check` for that